feat: Handle Directory Imports, Node Builtins

BeeeQueue opened the pull request:
This PR adds support for handling:

- Directory imports (`import x from "./directory"` where the file is `./directory/index.js`) 
- Node builtins (`import path from "path"`)

I first added tests to verify that these cases would be handled incorrectly, after which I fixed them 😄

Not sure if this is a `feat` or a `fix`

At least partially fixes #6

favna commented on src/index.ts:
```diff
@@ -63,6 +73,16 @@ function getFilter(options: PluginOptions): RegExp {
+let builtins: string[] | null = null;
+
+async function isBuiltin(path: string): Promise<boolean> {
+  if (builtins == null) {
```
```suggestion
  if (builtins === null) {
```

favna reviewed (changes requested):
I'm not yet sure why but this seems to fail on Windows (11, x64) where my snapshots update to invalid imports:

![image](https://github.com/favware/esbuild-plugin-file-path-extensions/assets/4019718/bdf3ba33-ca41-437e-b0a7-0a3529ccf207)

(Granted the old snapshots weren't correct either but the diff here should be the removal of `.cjs`)

BeeeQueue commented:
I found that most snapshots have that error, so I updated those as well.

I couldn't see why it wasn't using esbuild's `format` option as the default so I added that to fix said snapshots.

I still have no idea why the last snapshot is failing, even after going stepby-step in the debugger...

favna commented on tests/scenarios/node-builtins.test.ts:
```diff
@@ -0,0 +1,28 @@
+function createConfig(platform: 'node' | 'browser'): BuildOptions {
+  return createEsbuildConfig({
+    platform,
+    format: 'esm',
```
```suggestion
    format: 'esm',
    outExtension: {
      '.js': '.mjs'
    },
```

This will fix the last 2 tests. Currently it outputs as `.js` which conflicts with line 18 and 26.

favna commented:
I'll be honest I didn't realise you work for Paypal until I merged the PR and I saw your email in the commit body. That's pretty damn cool that my project is used at Paypal. That's pretty big for a simple open-source contributor like me. Thanks!!

favna commented:
Released in v2.1.0, not sure why GitHub doesn't detect it automatically.
